mesonbuild/interpreter/interpreter.py

```diff
@@ -2301,14 +2301,18 @@ def func_install_man(self, node: mparser.BaseNode,
         return m
 
     @FeatureNew('install_emptydir', '0.60.0')
+    @typed_pos_args('install_emptydir', varargs=str, min_varargs=1)
     @typed_kwargs(
         'install_emptydir',
         INSTALL_MODE_KW,
         KwargInfo('install_tag', (str, NoneType), since='0.62.0')
     )
-    def func_install_emptydir(self, node: mparser.BaseNode, args: T.Tuple[str], kwargs) -> None:
-        d = build.EmptyDir(args[0], kwargs['install_mode'], self.subproject, kwargs['install_tag'])
-        self.build.emptydir.append(d)
+    def func_install_emptydir(self, node: mparser.BaseNode, args: T.Tuple[T.List[str]], kwargs) -> None:
+        if len(args[0]) > 1:
+            FeatureNew.single_use('install_emptydir with more than one path', '1.4.0', self.subproject, location=node)
+        self.build.emptydir += \
+            [build.EmptyDir(one_dir, kwargs['install_mode'], self.subproject, kwargs['install_tag'])
+             for one_dir in args[0]]
```

Comment:
This is a complicated way to write:

```
for one_dir in args[0]:
    d = build.EmptyDir(one_dir, kwargs['install_mode'], self.subproject, kwargs['install_tag'])
    self.build.emptydir.append(d)
```

... which is also the style used by other functions.

Reply:
Will change if you want, but I should mention that I see nothing complicated in set-builder notation. AFAIK these set-comprehensions are pythonic way of coding. Though I personally don't consider myself a Python programmer that much, but I like the style because it's functional compared to one you suggested which is imperative.

Reply:
It is a list comprehension, not a set.

And list comprehensions are an extremely powerful, valuable, amazing feature of python -- and should be treated with care as a result. Here, a list comprehension is being used with two different forms of line continuation because the thing being done is so *long*. This does not enhance readability IMO.

Reply:
Done!